devito/at_controller.py

```diff
@@ -91,71 +92,69 @@ def auto_tune_blocks(self, minimum=5, maximum=20):
         at_nt = 3
         self.op.propagator.nt = at_nt
         self.op.propagator.profile = True
-        self.op.propagator.cache_blocking = [0 if dim else None
-                                             for dim in self.blocked_dims]
-        # want to run function at least once to make sure block_sizes are not
-        # overwritten
-        self.get_execution_time()
-        logger.info("Starting auto tuning of block sizes using brute force")
-
-        times = []  # list where times and block sizes will be kept
+
+        info_at("Start. Mode: brute force")
+
         block_list = set()  # used to make sure we do not test the same block sizes
-        blocks = list(self.op.propagator.block_sizes)
+        mask = [i if i else None for i in self.blocked_dims]
+        block = [None for i in mask]
 
         for x in range(minimum, maximum):
-            blocks[0] = x if blocks[0] else None
+            block[0] = mask[0] and x
 
-            if len(blocks) > 1:
+            if len(block) > 1:
                 for y in range(minimum, maximum):
-                    blocks[1] = y if blocks[1] else None
+                    block[1] = mask[1] and y
 
-                    if len(blocks) > 2:
+                    if len(block) > 2:
                         for z in range(minimum, maximum):
-                            blocks[2] = z if blocks[2] else None
-                            block_list.add((tuple(blocks)))
+                            block[2] = mask[2] and z
+                            block_list.add((tuple(block)))
                     else:
-                        block_list.add(tuple(blocks))
+                        block_list.add(tuple(block))
             else:
-                block_list.add(tuple(blocks))
+                block_list.add(tuple(block))
+
+        # filter off some of the block sizes, heuristically
+        block_list = sorted(self._filter(block_list))
 
-        logger.info("Total number of block size permutations = %d" % len(block_list))
-        if maximum - minimum > 2:  # No point to estimate if diff is below 2.
-            self._estimate_run_time(minimum, maximum, len(block_list))
+        # populate output arrays with random values, to make sure that loads
+        # and stores are performed
+        for param in self.op.output_params:
```

Comment:
Hmm, ok, you're right, the ones you're after are the RW ones, really. We don't have particularly sophisticated access descriptors yet, so I guess this is fine then.